[Metrics][Discover] Adds dimension type icons to metrics flyout

simianhacker opened the pull request:
## 🍒 Summary

This pull request adds support for different dimension types in the unified metrics grid flyout. It maps various dimension types to their corresponding `EuiToken` icons, providing a more informative and visually consistent user experience.

## 🛠️ Changes

- Maps dimension types to EuiToken icons
- Displays a token icon for boolean, ip, keyword, and numeric types

## 🎙️ Prompts

- "I just merged this PR: https://github.com/elastic/kibana/pull/235276 I would like to add support for the different dimension types defined in `src/platform/plugins/shared/metrics_experience/common/fields/constants.ts`. I would like to `EuiToken` to replace type of dimension in the list."
- "Here are the tokens and how they should map:"
- "`tokenBoolean`: boolean"
- "`tokenIP`: ip"
- "`tokenKeyword`: keyword"
- "`tokenNumber`: long, integer, short, byte, unsigned_long"

<img width="2047" height="1003" alt="image" src="https://github.com/user-attachments/assets/1a3af13b-f8bb-44ac-8d90-e260c6bb8786" />


🤖 This pull request was assisted by Gemini CLI

simianhacker committed:
Adds dimension type icons to metrics flyout

- Maps dimension types to EuiToken icons
- Displays a token icon for boolean, ip, keyword, and numeric types

Prompts:

- "I just merged this PR: https://github.com/elastic/kibana/pull/235276 I would like to add support for the different dimension types defined in `src/platform/plugins/shared/metrics_experience/common/fields/constants.ts`. I would like to EuiToken to reflaced type of dimension in the list."
- "Here are the tokens and how they should map:"
- "`tokenBoolean`: boolean"
- "`tokenIP`: ip"
- "`tokenKeyword`: keyword"
- "`tokenNumber`: long, integer, short, byte, unsigned_long"

🤖 This commit was assisted by Gemini CLI

elasticmachine commented:
Pinging @elastic/obs-ux-infra_services-team (Team:obs-ux-infra_services)

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/340534)
* Commit: 6f48d0c5556b7a108b83bb2acef283b92ba67916

### Failed CI Steps
* [FTR Configs #28](https://buildkite.com/elastic/kibana-pull-request/builds/340534#01995916-78c4-499f-a7b5-0c0213018255)
* [FTR Configs #41](https://buildkite.com/elastic/kibana-pull-request/builds/340534#01995916-78d0-4314-9215-b341ecd1d52f)
* [FTR Configs #88](https://buildkite.com/elastic/kibana-pull-request/builds/340534#01995916-78fb-46be-aa5b-1de0aa76cb1e)
* [Post-Build](https://buildkite.com/elastic/kibana-pull-request/builds/340534#019958f6-c830-4b78-a4f1-63d4a7c326be)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/340534#01995916-78fb-46be-aa5b-1de0aa76cb1e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/340534/jobs/01995916-78fb-46be-aa5b-1de0aa76cb1e/artifacts/01995943-4c1f-4a14-aa4a-65feb8c5dee3) FTR Configs #<span></span>88 / discover/security/context_awareness cell renderer ES|QL mode should render alert workflow status badge

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ‼️ ERROR: no builds found for mergeBase sha [340ca650bd4865e085bd99674e55df8a5542a103]

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340534","number":340534,"commit":"6f48d0c5556b7a108b83bb2acef283b92ba67916"}],"number":340534}
buildkite-pr-comment-->

simianhacker committed:
Merge branch 'main' of github.com:elastic/kibana into feature/add-dimension-type-icons

simianhacker committed:
Refactors icon mapping to use a Map

- Replaces the plain object icon mapping with a JavaScript Map
- Uses `Map.has()` to check for the existence of an icon
- Uses `Map.get()` to retrieve the icon type

🤖 This commit was assisted by Gemini CLI

jennypavlova reviewed (approved):
LGTM 🚀 Thanks for applying the suggestion! 

[Out of scope] We should make those a bit bigger in the redesign issue (there is already a new design there). The text in the icons is barely visible: 
<img width="540" height="331" alt="image" src="https://github.com/user-attachments/assets/56fb9a5f-c4b7-4ec8-ad33-0ffddfe1fc81" />
